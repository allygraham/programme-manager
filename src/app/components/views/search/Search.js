import React from 'react';
import PropTypes from 'prop-types';

export const Search = ({
  handleSearch, searchValue
}) => (
  <div className="field">
    <p className="control">
      <input
        type="search"
        onChange={(event) => handleSearch(event.target.value)}
        value={searchValue}
        className="input is-medium"
        placeholder="Search a name..."
      />
    </p>
  </div>
);

Search.propTypes = {
  handleSearch: PropTypes.func.isRequired,
  searchValue: PropTypes.string.isRequired,
};