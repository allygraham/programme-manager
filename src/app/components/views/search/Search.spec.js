import React from 'react';
import { shallow } from 'enzyme';

import { Search } from './Search';

describe('Search', () => {

  test('render the <Search> component', () => {
    const handleSearch = jest.fn();
    const clearSearch = jest.fn();
    const searchValue = '100';

    const wrapper = shallow(<Search handleSearch={handleSearch} clearSearch={clearSearch} searchValue={searchValue} />);
    expect(wrapper).toMatchSnapshot();
  });

});