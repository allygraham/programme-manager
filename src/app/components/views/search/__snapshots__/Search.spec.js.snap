// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search render the <Search> component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    clearSearch={[MockFunction]}
    handleSearch={[MockFunction]}
    searchValue="100"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p
        className="control"
      >
        <input
          className="input is-medium"
          onChange={[Function]}
          placeholder="Search a name..."
          type="search"
          value="100"
        />
      </p>,
      "className": "field",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          className="input is-medium"
          onChange={[Function]}
          placeholder="Search a name..."
          type="search"
          value="100"
        />,
        "className": "control",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input is-medium",
          "onChange": [Function],
          "placeholder": "Search a name...",
          "type": "search",
          "value": "100",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p
          className="control"
        >
          <input
            className="input is-medium"
            onChange={[Function]}
            placeholder="Search a name..."
            type="search"
            value="100"
          />
        </p>,
        "className": "field",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="input is-medium"
            onChange={[Function]}
            placeholder="Search a name..."
            type="search"
            value="100"
          />,
          "className": "control",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input is-medium",
            "onChange": [Function],
            "placeholder": "Search a name...",
            "type": "search",
            "value": "100",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
